src/parsing.jl

```diff
@@ -316,14 +316,43 @@ function pqparse(::Type{InfExtendedTime{T}}, str::AbstractString) where T<:Dates
 end
 
 # UNIX timestamps
-function Base.parse(::Type{DateTime}, pqv::PQTextValue{PQ_SYSTEM_TYPES[:int8]})
+function Base.parse(::Type{DateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:int8]})
     return unix2datetime(parse(Int64, pqv))
 end
 
-function Base.parse(::Type{ZonedDateTime}, pqv::PQTextValue{PQ_SYSTEM_TYPES[:int8]})
+function Base.parse(::Type{ZonedDateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:int8]})
     return TimeZones.unix2zdt(parse(Int64, pqv))
 end
 
+# All postgresql timestamptz are stored in UTC time with the epoch of 2000-01-01.
+const POSTGRES_EPOCH_DATE = Date("2000-01-01")
+
+# Note: Because postgresql stores the values as a Microsecond in Int64, the max (infinite)
+# value of date time in postgresql when querying binary is 294277-01-09T04:00:54.775
+# and the minimum is -290278-12-22T19:59:05.225.
+function pqparse(::Type{ZonedDateTime}, ptr::Ptr)
+    return ZonedDateTime(POSTGRES_EPOCH_DATE, tz"UTC") +
+        Microsecond(ntoh(unsafe_load(Ptr{Int64}(ptr))))
```

Comment:
Not BlueStyle

Reply:
This line can be made much faster if the `+` is moved to be between the `DateTime` and the `Milisecond`
as it avoids needing to use a bunch of logic for  handing timezone transitions etc.
(also likely because of manipulating ZonedDateTimes being slow due to them not being is bits)

```julia
using TimeZones, BenchmarkTools, Dates

const POSTGRES_EPOCH_DATE = Date("2000-01-01");
@btime ZonedDateTime(POSTGRES_EPOCH_DATE, tz"UTC") + Microsecond(ntoh(dat)) setup=(dat=bswap(rand(100:10_000_000)))
# 11.408 ns (0 allocations: 0 bytes)

const POSTGRES_EPOCH_UNIX = datetime2unix(DateTime("2000-01-01"));
@btime ZonedDateTime(unix2datetime(POSTGRES_EPOCH_UNIX + ntoh(dat)÷1000), tz"UTC") setup=(dat=bswap(rand(100:10_000_000)))
#     4.417 ns (0 allocations: 0 bytes)

const POSTGRES_EPOCH_DATETIME = DateTime("2000-01-01");
@btime ZonedDateTime(POSTGRES_EPOCH_DATETIME + Microsecond(ntoh(dat)), tz"UTC") setup=(dat=bswap(rand(100:10_000_000)))
#   1.509 ns (0 allocations: 0 bytes)
```